server/internal/api/apiv1/post_init_handlers.go

```diff
@@ -95,6 +95,11 @@ func (s *PostInitHandlers) GetJoinOptions(ctx context.Context, req *api.ClusterJ
 		return nil, apiErr(err)
 	}
 
+	// Validate that the host ID is unique in the cluster
+	if err := validateHostIDUniqueness(ctx, s.hostSvc, hostID); err != nil {
+		return nil, makeInvalidInputErr(err)
```

Comment:
If you take my advice below, you should change this to 

```suggestion
		return nil, apiErr(err)
```